FileProviderExtensionUI/FileProviderCoordinator.swift

```diff
@@ -63,6 +63,13 @@ class FileProviderCoordinator: Coordinator {
 
 	func handleError(_ error: Error, for viewController: UIViewController) {
 		DDLogError("Error: \(error)")
+		if let fileProviderError = error as? FileProviderCoordinatorError {
+			switch fileProviderError {
+			case let .unauthorized(vaultName):
+				showUnauthorizedError(vaultName: vaultName)
+				return
+			}
+		}
```

Comment:
I'm still thinking about a simplification here, because the switch statement only has one case. How about this?

```swift
func handleError(_ error: Error, for viewController: UIViewController) {
	DDLogError("Error: \(error)")
	if let fileProviderError = error as? FileProviderCoordinatorError, case let .unauthorized(vaultName) = fileProviderError {
		showUnauthorizedError(vaultName: vaultName)
	} else {
		let alertController = UIAlertController(title: LocalizedString.getValue("common.alert.error.title"), message: error.localizedDescription, preferredStyle: .alert)
		alertController.addAction(UIAlertAction(title: LocalizedString.getValue("common.button.ok"), style: .default))
		viewController.present(alertController, animated: true)
	}
}
```

Reply:
I’ve applied the change in commit b1e02eb, and it definitely makes the code cleaner and easier to read. Thanks again for the helpful input!